### PR TITLE
[Resource-Timing] Flakiness reduction

### PR DESCRIPTION
**Pull request by yoavweiss**
Alternative approach to #14616
Closes w3c/resource-timing#182 and #14433

Reducing RT test flakiness by:

* Adding an extra task before actual testing in buffer-full-add-after-full-event.html, as the timing of when entries are added to the buffer is not really well-defined at the moment.
* Remove use of `test_equals` in resource_initiator_types.html

### CODE REVIEW

**Comment by yoavweiss** [comment]
There seem to be some flakiness in Firefox nightly, but it doesn't seem to be the test's fault. Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1516024

File: resource-timing/buffer-full-add-after-full-event.html [hunk 1]
```diff
@@ -72,6 +72,9 @@
     await loadRandomResource();
     await waitForEventToFire();
     await clearAndAddAnotherEntryToBuffer();
+    // Since we have no strict guarantees when an entry will be added to the
+    // buffer, waiting till next task to try to avoid flakiness.
+    await waitTillNextTask();
```
Comment (cvazac) [inline resource-timing/buffer-full-add-after-full-event.html]: s/waitTillNextTask/waitForNextTask

Comment (yoavweiss) [reply resource-timing/buffer-full-add-after-full-event.html]: nm. I misunderstood your comment and missed my own typo. Fixing

**Review by cvazac — APPROVED** [review]

**Comment by Hexcles** [comment]
The flakiness on Firefox is likely a genuine implementation issue according to https://bugzilla.mozilla.org/show_bug.cgi?id=1516024 . Yoav will discuss this issue later in their WG meeting. Force-merging this PR.

**Comment by Hexcles** [comment]
Yikes, should've squashed instead of rebasing it. Sorry :(

**Pull request merged**